### PR TITLE
Port/Decompile pokedex.c and Dump Pokedex Entries and Text to C

### PR DESCRIPTION
**Pull request by ultima-soul**

### CODE REVIEW

File: ld_script.txt [hunk 1]
```diff
@@ -429,6 +429,8 @@ SECTIONS {
         data/data_83FECCC.o(.rodata.841EE44);
         src/new_menu_helpers.o(.rodata);
         data/data_83FECCC.o(.rodata.841F4B4);
+		src/pokedex.o(.rodata);
```
Comment (jiangzhengwenjz) [inline ld_script.txt]: You may want to change tab into space

File: src/pokedex.c [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+const u8 *sub_8088E20(u16 dexNum)
+{
+    return gPokedexEntries[dexNum].categoryName;
+}
+
+u16 GetPokedexHeightWeight(u16 dexNum, u8 data)
+{
+    switch (data)
+    {
+        case 0:  // height
```
Comment (jiangzhengwenjz) [inline src/pokedex.c]: Inconsistent switch-case indent. You did differently in GetNationalPokedexCount

File: src/pokedex.c [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+u16 GetKantoPokedexCount(u8 caseID)
+{
+    u16 count = 0;
+    u16 i;
+
+    for (i = 0; i < KANTO_DEX_COUNT; i++)
+    {
+        switch (caseID)
+        {
+            case FLAG_GET_SEEN:
```
Comment (jiangzhengwenjz) [inline src/pokedex.c]: Again

File: src/pokedex.c [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    for (i = 0; i < KANTO_DEX_COUNT - 1; i++)
+    {
+        if (!GetSetPokedexFlag(i + 1, FLAG_GET_CAUGHT))
+            return FALSE;
+    }
+    return TRUE;
+}
+
+u16 HasAllMons(void)
```
Comment (jiangzhengwenjz) [inline src/pokedex.c]: bool16. And use TRUE/FALSE macro.

**Comment by ultima-soul** [comment]
Resolved suggestions so far.

File: src/data/pokemon/pokedex_entries.h [hunk 5]
```diff
@@ -0,0 +1,5033 @@
+const struct PokedexEntry gPokedexEntries[] =
+{
+    [NATIONAL_DEX_NONE] =
+    {
+        .categoryName = _("UNKNOWN"),
+        .height = 0,
+        .weight = 0,
+        .description = gDummyPokedexText,
+        .unusedDescription = gDummyPokedexText + ARRAY_COUNT(gDummyPokedexText),
```
Comment (PikalaxALT) [inline src/data/pokemon/pokedex_entries.h]: Use gDummyPokedexTextUnused here

**Pull request merged**